### PR TITLE
dssim 1.3.2 (new formula)

### PR DESCRIPTION
**Pull request by swernerx**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Added Kornels `dssim` tool to compare RGBA structural similarity of different images / compressions.

### CODE REVIEW

File: Formula/dssim.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "libpng"
+
+  def install
+    system "make"
+    bin.install Dir["bin/*"]
+  end
+
+  test do
+    system "false"
```
Comment (mikemcquaid) [inline Formula/dssim.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Formula/dssim.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/pornel/dssim"
+  url "https://github.com/pornel/dssim/archive/1.1.tar.gz"
+  sha256 "bad5bde3846e499be16e551248168b98195ca1e5893f948739ecc0cf2e8d52f6"
+
+  depends_on "pkg-config" => :build
+  depends_on "libpng"
+
+  def install
+    system "make"
+    bin.install Dir["bin/*"]
```
Comment (mikemcquaid) [inline Formula/dssim.rb]: Is there no `make install` we can use? If not, could you file an issue asking upstream for one?

Comment (swernerx) [inline Formula/dssim.rb]: There is no "install" target in the makefile of that tool. What do you mean by asking "upstream"? It's really just a single binary.

File: Formula/dssim.rb (hunk 1 above)
Comment (swernerx) [inline Formula/dssim.rb]: I don't think there is a lot we can do here. For check whether it's working we would need two images to compare as far as I understand. I am not the author of the tool - I just want to use it :)

File: Formula/dssim.rb (hunk 2 above)
Comment (mikemcquaid) [inline Formula/dssim.rb]: I mean file an issue on `https://github.com/pornel/dssim` asking requesting (or, ideally, adding) a `make install` target so if more files should be installed in future we don't miss them. Also, if it's currently a single binary then `bin.install "bin/binary"` is preferable to globbing.

File: Formula/dssim.rb (hunk 1 above)
Comment (mikemcquaid) [inline Formula/dssim.rb]: We can check two images. Check out https://github.com/Homebrew/homebrew-core/blob/4c37b28a39c9cd066edb7024e9e75ab8d22ab7d4/Formula/imagemagick.rb#L128. A test is a requirement to merge this formula, unfortunately.

**Comment by ilovezfs** [comment]
https://github.com/pornel/dssim/blob/master/README.md

> On OS X make USE_COCOA=1 will compile without libpng.

You should probably make that the default, and then offer libpng as an optional dependency:
```
depends_on "libpng" => :optional
```

And then skip the `USE_COCOA=1` when it's `build.with? "libpng"`

**Comment by swernerx** [comment]
I tried first using without libpng without success. Therefor I used it as a normal dependency.

**Comment by UniqMartin** [comment]
Mostly looking good, but please address the following two audit issues:

```
dssim:
  * Stable: Use GitHub tarballs rather than zipballs (url is https://github.com/pornel/dssim/archive/1.3.2.zip).
  * Dir["bin/dssim"] is unnecessary; just use "bin/dssim"
```

(And as mentioned in the checklist in the first post, don't forget that you can and should check locally whether you have resolved those issues by running `brew audit --strict --online dssim`.)

**Comment by swernerx** [comment]
Fine. I changed the part and also checked using the audit tool.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**